resources/lib/tubecast/youtube/app.py

```diff
@@ -414,8 +436,7 @@ def __player_thread(self):
         self.player = CastPlayer(youtubecastv1=self)
         while not monitor.abortRequested() and self.has_client:
             monitor.waitForAbort(1)
-        # Del player
-        del self.player
```

Comment:
`del` doesn't "delete" the player, it just removes the binding from *self*. Because the player is set to `None` on the following line this doesn't seem necessary.

Reply:
Iirc this was placed here because player was a lot of times left in memory after abort was requested

Reply:
I see. I did notice that there was a log saying that there were classes left in memory. I thought they were related to something else but I will investigate if this line fixes it.

Reply:
You might be right. App holds the player and holds the request object. If Kodi hangs due to the keep-alive request, app is still alive and holding a reference to the player. Kodi will only warn about the classes it knows, in this case xbmc.Player. This might not be needed anymore if app shuts down gracefully.

Reply:
But better to confirm

Reply:
Indeed this doesn't seem to be the cause of the leftover classes.

Some observations:
- Only occurs when a device is connected during shutdown.
- Kodi reports `N14PythonBindings31XBMCAddon_xbmc_Monitor_DirectorE`  to be the class left in memory. I'm not familiar enough with Kodi's internals to properly interpret this but to me it seems as though this would be related to `kodi.Monitor()` somehow.

Reply:
Should I go ahead and open a separate issue for this? The warning is also present in the master branch so it isn't be related to this line or this pull request.

I don't think this is a top priority issue but I am very curious as to what causes this and it is still a memory leak.

Reply:
Sure go ahead